sacrebleu.py

```diff
@@ -656,6 +656,46 @@
+        'abcnews.306758=OTHER-politics-MX cnbc.com.6772=US-business reuters.218932=OTHER-politics-MK upi.176251=GB-sport reuters.218921=US-sport cnn.304447=US-politics '
+        'guardian.221679=GB-politics scotsman.133765=GB-sport scotsman.133804=GB-entertainment guardian.221762=OTHER-politics-BO cnbc.com.6769=US-politics '
+        'dailymail.co.uk.298692=EU-entertainment scotsman.133744=GB-world reuters.218911=US-sport newsweek.51310=US-politics independent.226301=US-sport reuters.218923=EU-sport '
+        'reuters.218861=US-politics dailymail.co.uk.298759=US-world scotsman.133791=GB-sport cbsnews.248484=EU-scitech dailymail.co.uk.298630=US-scitech '
+        'newsweek.51329=US-entertainment bbc.381701=GB-crime dailymail.co.uk.298738=GB-entertainment bbc.381669=OTHER-world-CN foxnews.94512=US-politics '
+        'guardian.221718=GB-entertainment dailymail.co.uk.298686=GB-politics cbsnews.248471=US-politics newsweek.51318=US-entertainment rt.com.91335=US-politics '
+        'newsweek.51300=US-politics cnn.304478=US-politics upi.176275=US-politics telegraph.405422=OTHER-world-ID reuters.218933=US-politics newsweek.51328=US-politics '
+        'newsweek.51307=US-business bbc.381692=GB-world independent.226346=GB-entertainment bbc.381646=GB-sport reuters.218914=US-sport scotsman.133758=EU-sport '
+        'rt.com.91350=EU-world scotsman.133773=GB-scitech rt.com.91334=EU-crime bbc.381680=GB-politics guardian.221756=US-politics scotsman.133783=GB-politics cnn.304521=US-sport '
+        'dailymail.co.uk.298622=GB-politics bbc.381789=GB-sport dailymail.co.uk.298644=GB-business dailymail.co.uk.298602=GB-world scotsman.133753=GB-sport '
+        'independent.226317=GB-entertainment nytimes.184862=US-politics thelocal.65969=OTHER-world-SY nytimes.184825=US-politics cnbc.com.6784=US-politics nytimes.184804=US-politics '
+        'nytimes.184830=US-politics scotsman.133801=GB-sport cnbc.com.6770=US-business bbc.381760=GB-crime reuters.218865=OTHER-world-ID newsweek.51339=US-crime '
+        'euronews-en.153797=OTHER-world-ID abcnews.306774=US-crime dailymail.co.uk.298696=GB-politics abcnews.306755=US-politics reuters.218909=US-crime '
+        'independent.226349=OTHER-sport-RU newsweek.51330=US-politics bbc.381705=GB-sport newsweek.51340=OTHER-world-ID cbsnews.248411=OTHER-world-FM abcnews.306776=US-crime '
+        'bbc.381694=GB-entertainment rt.com.91356=US-world telegraph.405430=GB-entertainment telegraph.405404=EU-world bbc.381749=GB-world telegraph.405413=US-politics '
+        'bbc.381736=OTHER-politics-KP cbsnews.248394=US-politics nytimes.184822=US-world telegraph.405408=US-politics euronews-en.153799=OTHER-politics-SY '
+        'euronews-en.153826=EU-sport cnn.304400=US-world'
+}
+SUBSETS = {k:{d.split('=')[0]:d.split('=')[1] for d in v.split()} for (k,v) in SUBSETS.items()}
+COUNTRIES = sorted(list({v.split('-')[0] for v in SUBSETS['wmt19'].values()}))
+DOMAINS = sorted(list({v.split('-')[1] for v in SUBSETS['wmt19'].values()}))
 
```

Comment:
I think this would be a lot more readable (and recreatable) if you encoded the dictionary format directly, e.g.,

```python
'wmt19': {
  'GB-politics': ['bbc.381790', 'guardian.221679', '...'],
  '...'
}
```

Reply:
The mapping should be docID->annotation, not the opposite way because:
- It is used this way when filtering the test set. I iterate through the test set and check whether a given annotation (e.g `GB-politics`) matches a given **regular expression** specified with `--subset`, so I cannot use mapping you suggest (annotation->list(docID)). I could use it just for specifying the annotation in the source code and build the docID->annotation mapping on the fly, but this seems as an unnecessary obfuscation.
- It is more readable and editable/extensible I think. E.g. it is easy to add more tags to a given docID (the documents are listed in the order they appear in the SGM file).
- My annotation includes also fine-grained country codes, e.g. in `OTHER-politics-MX`, the `MX` means Mexico. So for many annotation strings there would be just a single docID and the source code would be actually longer than now.

While reviewing this code, I admit there is a potential problem that the list of `COUNTRIES` and `DOMAINS` is based on `wmt19` only. Currently `wmt19` countries and domains are a superset of those from `wmt18` and no other test set is annotated, so I opted for keeping the code simple. It can be easily updated when anyone adds annotation for other test-sets with extra domains or countries (although, I consider the current inventory exhaustive as it includes country=OTHER and domain=world, which kind of also means "other").